app/navigation/NavigatorForChangeActivity.scala

```diff
@@ -119,34 +120,53 @@ class NavigatorForChangeActivity @Inject() extends Navigator {
     case _ => _ => defaultCall
   }
 
-  private def navigationFollowingImports(userAnswers: UserAnswers): Call = {
-    userAnswers.get(AmountProducedPage) match {
-      case Some(AmountProduced.Large) => navigateForLargeAmountProducedFollowingImports(userAnswers)
-      case Some(AmountProduced.None) => navigateForAmountProducedNoneFollowingImports(userAnswers)
-      case Some(AmountProduced.Small) => defaultCall
+  private def navigationFollowingImports(userAnswers: UserAnswers, mode: Mode): Call = {
+    (userAnswers.get(AmountProducedPage) , mode) match {
+      case (Some(AmountProduced.Large), mode) => navigateForLargeAmountProducedFollowingImports(userAnswers, mode)
+      case (Some(AmountProduced.None ), mode) => navigateForAmountProducedNoneFollowingImports(userAnswers, mode)
+      case (Some(AmountProduced.Small), mode) => navigateForAmountProducedSmallProducerImports(userAnswers, mode)
       case _ => routes.AmountProducedController.onPageLoad(NormalMode)
     }
   }
 
-  private def navigateForAmountProducedNoneFollowingImports(userAnswers: UserAnswers): Call =
-    userAnswers.get(ContractPackingPage)
+  private def navigateForAmountProducedSmallProducerImports(userAnswers: UserAnswers, mode: Mode): Call = {
+    (userAnswers.get(SecondaryWarehouseDetailsPage), mode)
     match {
-      case Some(true) if userAnswers.packagingSiteList.isEmpty =>
+      case (Some(_), mode)  if mode == CheckMode   => routes.ChangeActivityCYAController.onPageLoad
+      case (_ ,  mode) if mode == CheckMode   => routes.SecondaryWarehouseDetailsController.onPageLoad
+      case (_, _) => defaultCall
```

Comment:
What should happen in this case? I feel like it probably shouldn't still be defaultCall

Reply:
This is the navigation for when a small user get's to this page, I only done the navigation for check answers not sure if there's another ticket for the normal navigation or whether it should have been done before as the other cases were covered for when the producer was large or none